### PR TITLE
 Test cases for country model

### PR DESCRIPTION
**Pull request by suhasdombale123**
## Why?
To ensure that all the functions in country model are tested 

## This change addresses the need by:

added tests in model/country_test.exs file

[delivers #163144268](https://www.pivotaltracker.com/story/show/163144268)

## Checklist

- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

### CODE REVIEW

File: apps/snitch_core/test/data/model/country_test.ex [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      country = countries |> List.first()
+
+      country_list = Country.get(%{numcode: country.numcode})
+      assert country_list == country
+    end
+
+    test "fails, Why? value not present for a given key", %{countries: countries} do
+      country = countries |> List.first()
+
+      country_list = Country.get(%{numcode: "842"})
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/model/country_test.ex]: We can test without fetching the first country from the list and with the actual value

```elixir
assert Country.get(%{numcode: "842"}) == nil
```

File: apps/snitch_core/test/data/model/country_test.ex [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      refute country_list == country
+    end
+
+    test "for getting all records ", %{countries: countries} do
+      country = countries |> List.first()
+
+      country_list = Country.get_all()
+      assert country_list|> List.first() == country
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/model/country_test.ex]: Can we assert the length instead of the first element.

**Review by SagarKarwande — CHANGES_REQUESTED** [review]
Run elixir formatter.

File: apps/snitch_core/test/data/model/country_test.ex [hunk 3]
```diff
@@ -0,0 +1,39 @@
+defmodule Snitch.Data.Model.CountryTest do
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+  alias Snitch.Data.Model.Country
+  import Snitch.Factory
+
+  setup :countries
+
+  describe "Countries" do
```
Comment (arjun289) [inline apps/snitch_core/test/data/model/country_test.ex]: `describe` is usually used to group tests related to a particular type. 
e.g. It would be better to group tests by `functions` in the model. Here single describe has no purpose since all the tests are within this.

File: apps/snitch_core/test/data/model/country_test.ex [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    end
+
+    test "for getting records by a particular key ", %{countries: countries} do
+      country = countries |> List.first()
+
+      country_list = Country.get(%{numcode: country.numcode})
+      assert country_list == country
+    end
+
+    test "fails, Why? value not present for a given key", %{countries: countries} do
```
Comment (arjun289) [inline apps/snitch_core/test/data/model/country_test.ex]: If you look at test case calls, the statement generated won't make any sense. What fails here? If you group it by function name 
```
describe `get/1` do
   test "fails" do
   end

   test "succeeds" do
   end
end
```

This would make for more meaningful statements.

File: apps/snitch_core/test/data/model/country_test.ex [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      country = countries |> List.first()
+
+      country_list = Country.get(%{numcode: "842"})
+      refute country_list == country
+    end
+
+    test "for getting all records ", %{countries: countries} do
+      country = countries |> List.first()
```
Comment (arjun289) [inline apps/snitch_core/test/data/model/country_test.ex]: `List.first()` may fail since the order is not defined in `get_all`

**Review by arjun289 — CHANGES_REQUESTED** [review]
Try and group together tests in `describe` for different functions, it will also allow grouping edge cases better.

**Review by SagarKarwande — APPROVED** [review]
